.travis.yml

```diff
@@ -0,0 +1,68 @@
+language: c
+dist: trusty
+sudo: false
+group: beta
+
+# To cache doc-building dependencies.
+cache: pip
+
+branches:
+  only:
+    - master
+    - /^\d\.\d$/
+
+matrix:
+  fast_finish: true
+  allow_failures:
+    - env: OPTIONAL=true
+  include:
+    - os: linux
+      language: c
+      compiler: clang
+      # gcc also works, but to keep the # of concurrent builds down, we use one C
+      # compiler here and the other to run the coverage build. Clang is preferred
+      # in this instance for its better error messages.
+      env: TESTING=cpython
+    - os: osx
+      language: c
+      compiler: clang
+      # Testing under macOS is optional until testing stability has been demonstrated.
+      env: OPTIONAL=true
+      before_install:
+        - brew install openssl xz
+        - export CPPFLAGS="-I$(brew --prefix openssl)/include"
+        - export LDFLAGS="-L$(brew --prefix openssl)/lib"
+
+# Travis provides only 2 cores, so don't overdo the parallelism and waste memory.
+before_script:
+  - |
+      set -e
+      if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.rst$)|(^Doc)|(^Misc)'
+      then
+        echo "Only docs were updated, stopping build process."
+        exit
+      fi
+      ./configure --with-pydebug
+      make -j4
```

Comment:
We should perhaps keep the `make clinic` check.  That might require also backporting `.gitattributes` (which might not be a bad idea anyway).